src/sync.js

```diff
@@ -30,136 +30,201 @@ function decrypt(key, encrypted) {
   });
 }
 
+// An "id schema" used to validate Kinto IDs and generate new ones.
+const notesIdSchema = {
+  // FIXME: Maybe this should generate IDs?
+  generate() {
+    throw new Error("cannot generate IDs");
+  },
 
-function loadFromKinto(client) {
+  validate(id) {
+    // FIXME: verify that at least this matches Kinto server ID format
+    return true;
+  },
+};
+
+class ServerKeyNewerError extends Error {
+  constructor() {
+    super("key used to encrypt the record appears to be newer than our key");
+  }
+}
+
+class ServerKeyOlderError extends Error {
+  constructor() {
+    super("key used to encrypt the record appears to be older than our key");
+  }
+}
+
+class JWETransformer {
+  constructor(key) {
+    this.key = key;
+  }
+
+  async encode(record) {
+    // FIXME: should we try to obfuscate the record ID?
+    const ciphertext = await encrypt(this.key, record);
+    // Copy over the _status field, so that we handle concurrency
+    // headers (If-Match, If-None-Match) correctly.
+    // DON'T copy over "deleted" status, because then we'd leak
+    // plaintext deletes.
+    const status = record._status && (record._status == "deleted" ? "updated" : record._status);
+    const encryptedResult = {
+      content: ciphertext,
+      id: record.id,
+      _status: status,
+      kid: this.key.kid,
+    };
+    if (record.hasOwnProperty("last_modified")) {
+      encryptedResult.last_modified = record.last_modified;
+    }
+
+    return encryptedResult;
+  }
+
+  async decode(record) {
```

Comment:
Same here.. should this be `async decode(record) {}` or `async function decode(record) {}`?